pkg/util/translate/translate.go

```diff
@@ -65,6 +67,7 @@ func GetOwnerReference() []metav1.OwnerReference {
 			Kind:       typeAccessor.GetKind(),
 			Name:       Owner.GetName(),
 			UID:        Owner.GetUID(),
+			Controller: &t,
```

Comment:
I think it should be set to `true` only if the virtual pod resource has some owner reference with `controller==true`.

btw: we are already checking for almost that here - [pkg/controllers/resources/pods/translate/translator.go#L154-L168](https://github.com/loft-sh/vcluster/blob/0403fecea6d3aea9fd53fddb9f02cd1cfd116fbb/pkg/controllers/resources/pods/translate/translator.go#L154-L168) 
That^ can be simplified with the use of `metav1.GetControllerOf(&pod)`
like it's used in the kubectl drain - https://github.com/kubernetes/kubernetes/blob/2af53e9f64eee81a09fd0fd1f0fc96ce0859eb0f/staging/src/k8s.io/kubectl/pkg/drain/filters.go#L237-L244